waspc/data/Generator/templates/server/src/routes/auth/signup.js

```diff
@@ -1,11 +1,30 @@
 {{={= =}=}}
+import Prisma from '@prisma/client'
 import prisma from '../../dbClient.js'
 import { handleRejection } from '../../utils.js'
+import AuthError from '../../core/AuthError.js'
+import HttpError from '../../core/HttpError.js'
 
 export default handleRejection(async (req, res) => {
   const userFields = req.body || {}
 
-  await prisma.{= userEntityLower =}.create({ data: userFields })
+  try {
+    await prisma.{= userEntityLower =}.create({ data: userFields })
+  } catch(e) {
+    if (e instanceof AuthError) {
+      throw new HttpError(422, 'Validation failed', { message: e.message })
+    } else if (e instanceof Prisma.PrismaClientKnownRequestError && e.code === 'P2002') {
+      // Warn about unique constraint for email already existing
+      throw new HttpError(422, 'Save failed',
+        {
```

Comment:
```suggestion
      throw new HttpError(422, 'Save failed', {
```

Reply:
Good call, I need to tune/wire up my formatter. Thnx!

Reply:
Fixed with refactor here: https://github.com/wasp-lang/wasp/pull/376/commits/08372fdbec58644bff09b7fa58bde2420071d9f2 I need to get VSCode JS formatter playing nicely with template syntax sometime this week. Seems to not even want to auto format only changes on save 🤔

Reply:
I think that will be very hard, I wouldn't bother with it actually. I also don't have it set up. And to be honest formatting here is not super important, so no worries for now.